packages/service/ni_measurement_plugin_sdk_service/discovery/_types.py

```diff
@@ -19,3 +26,18 @@ def insecure_address(self) -> str:
     def ssl_authenticated_address(self) -> str:
         """Get the service's SSL-authenticated address in the format host:port."""
         return f"{self.location}:{self.ssl_authenticated_port}"
+
+    @classmethod
+    def _from_grpc(cls, other: discovery_service_pb2.ServiceLocation) -> ServiceLocation:
+        return ServiceLocation(
+            location=other.location,
+            insecure_port=other.insecure_port,
+            ssl_authenticated_port=other.ssl_authenticated_port,
+        )
+
+
+class ServiceDetails(typing.NamedTuple):
+    """Represents the service information along with its location."""
+
+    service_location: ServiceLocation
+    service_info: ServiceInfo
```

Comment:
"details" and "info" mean about the same thing, and the other versions of this API (C# and LV) don't have a "service details" struct, so I think this named tuple makes the Python API more confusing. Use a normal tuple. It's ok for Python methods to return more than one thing.

Reply:
Sure, I have removed `ServiceDetails`.